auth.py

```
@@ -0,0 +1,200 @@
+    def __init__(
+        self,
+        session: aiohttp.ClientSession,
+        host: str,
+    ) -> None:
+        """Initialize the auth client."""
+        self.session = session
+        self.host = host
+        self.base_url = f"http://{host}"
+        self.session_id = None
+        
+    async def authenticate_with_access_code(self, access_code: str) -> str | None:
```

Ruff (0.12.2)

`authenticate_with_access_code` is too complex (25 > 10)

(C901)